src/prefect/cli/work_pool.py

```diff
@@ -380,6 +380,58 @@ async def update(
         exit_with_success(f"Updated work pool {name!r}")
 
 
+@work_pool_app.command()
+async def provision_infra(
+    name: str = typer.Argument(
+        ..., help="The name of the work pool to provision infrastructure for."
+    ),
+):
+    """
+    Provision infrastructure for a work pool.
+
+    \b
+    Examples:
+        $ prefect work-pool provision-infrastructure "my-pool"
```

Comment:
This example doesn't match the sub-command name, but I think it would be nice to have an alias for `provision-infrastructure` so that users can use either.

Reply:
It seems that Typer doesn't natively support command aliases so I extended our `PrefectTyper` implementation.

Reply:
Nice!